Example/SimpleDemo/NetworkStackDemoTests/RenewTokenTests.swift

```diff
@@ -0,0 +1,289 @@
+                                              route: route,
+                                              needsAuthorization: true)
+
+
+    var counter: Int = 0
+    var counter200: Int = 0
+    var counter401: Int = 0
+    stub(condition: pathEndsWith(route.path)) { _ in
+      let stubPath = OHPathForFile("empty.json", type(of: self))
+      counter += 1
+      // First time return 401, second time return 200 OK
+      if counter > 1 {
+        counter200 += 1
+        return fixture(filePath: stubPath!, status: 200, headers: ["Content-Type":"application/json"])
+      } else {
+        counter401 += 1
+        return fixture(filePath: stubPath!, status: 401, headers: ["Content-Type":"application/json"])
+      }
+    }
+
+    _ = networkStack.sendRequestWithJSONResponse(requestParameters: requestParameters)
+      .subscribe(onNext: { _ in
+        promise1.fulfill()
+      }, onError: { (error) in
+        XCTFail("Encounter error : \(error)")
+      })
+
+    _ = networkStack.sendRequestWithJSONResponse(requestParameters: requestParameters)
+      .subscribe(onNext: { _ in
+        promise2.fulfill()
+      }, onError: { (error) in
+        XCTFail("Encounter error : \(error)")
+      })
+
+    waitForExpectations(timeout: kTimeout, handler: { _ in
+      XCTAssertEqual(counter, 3)
+      XCTAssertEqual(counter200, 2)
+      XCTAssertEqual(counter401, 1)
+
+      // Check the token value
+      let token = self.networkStack.currentAccessToken()
+      XCTAssertNotNil(token)
+      XCTAssertEqual(token, "fakeRenewedToken1")
+    })
+  }
+
+  func testMultiRequestWithHTTP401() {
+    if let token = self.networkStack.currentAccessToken() {
+      XCTAssertEqual(token, initToken)
+    }
+
+    let promise1 = expectation(description: "testMultiRequestWithHTTP401 Request 1")
+    let promise2 = expectation(description: "testMultiRequestWithHTTP401 Request 2")
+    let promise3 = expectation(description: "testMultiRequestWithHTTP401 Request 3")
+
+    let route: TestRoute = TestRoute("/test")
+    let requestParameters = RequestParameters(method: .get,
+                                              route: route,
+                                              needsAuthorization: true)
+
+
+    var counter: Int = 0
+    var counter200: Int = 0
+    var counter401: Int = 0
+    stub(condition: pathEndsWith(route.path)) { _ in
+      let stubPath = OHPathForFile("empty.json", type(of: self))
+      counter += 1
+      // First time return 401, second time return 200 OK
+      if counter > 1 {
+        counter200 += 1
+        return fixture(filePath: stubPath!, status: 200, headers: ["Content-Type":"application/json"])
+      } else {
+        counter401 += 1
+        return fixture(filePath: stubPath!, status: 401, headers: ["Content-Type":"application/json"])
+      }
+    }
+
+    var nextCount: Int = 0
+    _ = networkStack.sendRequestWithJSONResponse(requestParameters: requestParameters)
```

Comment:
(This comment applies everywhere you used `_ = networkStack.send…`)

Don't forget to store the `Disposable` in a `DisposeBag` and clear the `DisposeBag` inside `setUp()` and/or `tearDown()`. This way, in case the test fails for timeout (the request, for some unknown reason e.g. because the test is broken, takes ages to response — even if it should be stubbed but maybe there's a bug in your test and it isn't), we'll cancel the requests after the test finish with failure.